### PR TITLE
feat(mme): AMBR Extended-2 field addition in EPS QoS IE and in APN-AMBR.

### LINKED ISSUE #5675: [mme + sessiond] Add 5G NSA IE support in Create Session Request
This issue is aimed to add support for 5G-NSA support in Magma MME and Sessiond (without any federated network)

1. Extend the [QosInformationRequest](https://github.com/magma/magma/blob/master/lte/protos/session_manager.proto#L510) to include units for AMBR, i.e. `bps` for 4G and `kbps` for 5G-NSA
2. Add logic at MME to include units in Create Session Request message, based on the IEs in Update Location Answer
3. Add logic at MME to include higher bitrates in Activate Default Bearer Request to UE with the **Extended UE-AMBR Downlink
and Uplink** IEs 
4. Add logic at Sessiond to parse the units field in the Create Session Request message

Ref: [LTE 5G Dual Connectivity Call Flow](http://www.eventhelix.com/5G/non-standalone-access-en-dc/en-dc-secondary-node-addition-core-network-details.pdf)

### PR DESCRIPTION
**Pull request by Vyas20**
[Converged-MME]AMBR Extended-2 field addition in NAS layer for EPS QoS IE and in APN-AMBR IE .

This is used to accommodate AMBR values greater than 256 Mbps in both uplink and downlink direction.

This PR is tested using S1AP tester test suite and dsTester.

Wirehsark trace with all the octets used is attached with this email.
[AMBRvalueAllOctets.zip](https://github.com/magma/magma/files/7005964/AMBRvalueAllOctets.zip)
[AMBRvalueAllOctets_1stSept.zip](https://github.com/magma/magma/files/7089457/AMBRvalueAllOctets_1stSept.zip)

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 1]
```diff
@@ -86,22 +86,22 @@ int encode_apn_aggregate_maximum_bit_rate(
   encoded++;
 
   if (apnaggregatemaximumbitrate->extensions &
-      APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT) {
+          APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT |
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: can you add unit test? see the example in directory https://github.com/magma/magma/tree/master/lte/gateway/c/core/oai/test/mme_app_task

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 2]
```diff
@@ -112,53 +112,119 @@ int encode_apn_aggregate_maximum_bit_rate(
 void bit_rate_value_to_eps_qos(
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c:118]: add unit test

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c [hunk 3]
```diff
@@ -88,6 +88,20 @@ status_code_e decode_eps_quality_of_service(
     epsqualityofservice->bitRatesExtPresent = 0;
   }
 
+  if (ielen > 10 + (iei > 0) ? 1 : 0) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c:93]: add unit test

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c [hunk 4]
```diff
@@ -173,6 +192,19 @@ status_code_e eps_qos_bit_rate_ext_value(uint8_t br) {
   }
 }
 
+//------------------------------------------------------------------------------
+status_code_e eps_qos_bit_rate_ext2_value(uint8_t br) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c]: add unit test

**Review by ulaskozat — CHANGES_REQUESTED** [review]
Please add unit tests.

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c (hunk 4 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c]: I added this code because there exists similar functions for other octets but these functions are not being used anywhere. Can you please confirm if we are using "eps_qos_bit_rate_value", "eps_qos_bit_rate_ext_value" anywhere ?

File: lte/gateway/c/core/oai/tasks/nas/esm/sap/esm_send.c [hunk 5]
```diff
@@ -279,6 +279,23 @@ status_code_e esm_send_activate_default_eps_bearer_context_request(
...
+        LOG_NAS_ESM, "ESM-SAP   - epsqos  maxBitRateForDL  Ext: %u\n",
+        qos->bitRatesExt2.maxBitRateForDL);
+    OAILOG_DEBUG(
+        LOG_NAS_ESM, "ESM-SAP   - epsqos  guarBitRateForUL Ext: %u\n",
+        qos->bitRatesExt2.guarBitRateForUL);
+    OAILOG_DEBUG(
+        LOG_NAS_ESM, "ESM-SAP   - epsqos  guarBitRateForDL Ext: %u\n",
+        qos->bitRatesExt2.guarBitRateForDL);
+  } else {
+    OAILOG_DEBUG(LOG_NAS_ESM, "ESM-SAP   - epsqos  no bit rates ext defined\n");
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/esm/sap/esm_send.c]: ext -> ext2

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 6]
```diff
@@ -112,53 +112,121 @@ int encode_apn_aggregate_maximum_bit_rate(
...
     ApnAggregateMaximumBitRate* apn_ambr, uint64_t ambr_dl, uint64_t ambr_ul) {
   uint64_t ambr_dl_kbps = ambr_dl / 1000;  // ambr_dl is expected in bps
-  uint64_t ambr_ul_kbps = ambr_ul / 1000;  // ambr_ul is expected in bps
-  if (ambr_dl_kbps == 0) {
+  uint64_t ambr_ul_kbps = ambr_ul / 1000;  // ambr_ul is expected in bps 8640000
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: what is the relevance of `8640000` in this comment?

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 7]
```diff
@@ -112,53 +112,121 @@ int encode_apn_aggregate_maximum_bit_rate(
...
+      ((ambr_dl_kbps >= 8600) && (ambr_dl_kbps <= 16000))) {
+    apn_ambr->extensions |=
+        APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
+    apn_ambr->apnambrfordownlink_extended = (ambr_dl_kbps - 8600) / 100;
+
+    ambr_dl_kbps =
+        ambr_dl_kbps - (apn_ambr->apnambrfordownlink_extended) * 100 - 8600;
+  }
+
+  else if ((ambr_dl_kbps > 16000) && (ambr_dl_kbps <= 128000)) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: apply linter here to fix formatting.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 8]
```diff
@@ -112,53 +112,121 @@ int encode_apn_aggregate_maximum_bit_rate(
...
+    } else if ((ambr_ul_kbps > 63) && (ambr_ul_kbps <= 575)) {
+      apn_ambr->apnambrforuplink = ((ambr_ul_kbps - 64) / 8) + 64;
+      ambr_ul_kbps =
+          ambr_ul_kbps - (apn_ambr->apnambrforuplink_extended - 64) * 8 - 63;
+    } else if ((ambr_ul_kbps > 575) && (ambr_ul_kbps <= 8640))
+
+    {
+      apn_ambr->apnambrforuplink = ((ambr_ul_kbps - 576) / 64) + 128;
+      ambr_ul_kbps =
+          ambr_ul_kbps - (apn_ambr->apnambrforuplink_extended - 128) * 64 - 575;
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: apply linter to fix formatting. If you run `make build_oai`, it should automatically format it. Please make sure that correctly formatted files are added/committed and pushed.

File: lte/gateway/c/core/oai/test/mme_app_task/test_EPSQualityOfService.cpp [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+    eps_qos_bit_rate_ext2_value_test) {
+    int eps_qos_bit_rate_ext2_value(uint8_t br);
+    ASSERT_EQ(eps_qos_bit_rate_ext2_value (0b00001010), 303104);
+   ASSERT_EQ(eps_qos_bit_rate_ext2_value (0b10011101), 1495040);
+    ASSERT_EQ(eps_qos_bit_rate_ext2_value (0b11110110), 10240000);
+    }
+
+int main(int argc, char** argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+  OAILOG_INIT("MME", OAILOG_LEVEL_DEBUG, MAX_LOG_PROTOS);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/mme_app_task/test_EPSQualityOfService.cpp]: I don't think you need this here.

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c (hunk 3 above)
Comment (ulaskozat) [reply lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c:93]: how did you test this?

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 10]
```diff
@@ -112,53 +112,121 @@ int encode_apn_aggregate_maximum_bit_rate(
...
-    apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 64) / 8) + 64;
-  } else if ((ambr_dl_kbps > 575) && (ambr_dl_kbps <= 8640)) {
-    apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 576) / 64) + 128;
-  } else if (ambr_dl_kbps > 8640) {
-    apn_ambr->apnambrfordownlink = 0xfe;
-    apn_ambr->extensions =
+  }
+
+  if ((ambr_dl_kbps > 8640) &&
+      ((ambr_dl_kbps >= 8600) && (ambr_dl_kbps <= 16000))) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: fix these conditions.

**Comment by ulaskozat** [comment]
can you attach pcaps also to show correct encoding happens on wire?

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 1 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: Unit test added.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 2 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c:118]: Unit test added.

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c (hunk 3 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c:93]: Unit test added.

File: lte/gateway/c/core/oai/tasks/nas/esm/sap/esm_send.c (hunk 5 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/esm/sap/esm_send.c]: Updated.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 6 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: Updated.

File: lte/gateway/c/core/oai/test/mme_app_task/test_EPSQualityOfService.cpp (hunk 9 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/test/mme_app_task/test_EPSQualityOfService.cpp]: Updated.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 11]
```diff
@@ -113,52 +118,120 @@ void bit_rate_value_to_eps_qos(
...
+      ((ambr_dl_kbps > 8600) && (ambr_dl_kbps <= 16000))) {
+    apn_ambr->extensions |=
+        APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
+    apn_ambr->apnambrfordownlink_extended = (ambr_dl_kbps - 8600) / 100;
+
+    ambr_dl_kbps =
+        ambr_dl_kbps - (apn_ambr->apnambrfordownlink_extended) * 100 - 8600;
+  }
+
+  else if ((ambr_dl_kbps > 16000) && (ambr_dl_kbps <= 128000)) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: I commented on this before, please format these files, such breaks between if and else if's should not be there.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 12]
```diff
@@ -113,52 +118,120 @@ void bit_rate_value_to_eps_qos(
...
-          ((ambr_dl_kbps - 128000) / 2000) + 186;
+    apn_ambr->apnambrfordownlink_extended =
+        ((ambr_dl_kbps - 128000) / 2000) + 186;
+
+    ambr_dl_kbps = ambr_dl_kbps -
+                   (apn_ambr->apnambrfordownlink_extended - 186) * 2000 -
+                   128000;
+  }
+
+  if ((apn_ambr->apnambrfordownlink != 0xfe) && !(ambr_dl_kbps > 8640)) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c:168]: `!(ambr_dl_kbps > 8640)` -> `(ambr_dl_kbps <= 8640)`

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 13]
```diff
@@ -113,52 +118,120 @@ void bit_rate_value_to_eps_qos(
...
+    if ((ambr_dl_kbps > 0) && (ambr_dl_kbps <= 63)) {
+      apn_ambr->apnambrfordownlink = ambr_dl_kbps;
+    } else if ((ambr_dl_kbps > 63) && (ambr_dl_kbps <= 575)) {
+      apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 64) / 8) + 64;
+      ambr_dl_kbps =
+          ambr_dl_kbps - (apn_ambr->apnambrfordownlink_extended - 64) * 8 - 63;
+    } else if ((ambr_dl_kbps > 575) && (ambr_dl_kbps <= 8640)) {
+      apn_ambr->apnambrfordownlink = ((ambr_dl_kbps - 576) / 64) + 128;
+      ambr_dl_kbps                 = ambr_dl_kbps -
+                     (apn_ambr->apnambrfordownlink_extended - 128) * 64 - 575;
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c:181]: please run `make build_oai` before adding files and committing to fix clang formatting issues.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 12 above)
Comment (ulaskozat) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c:168]: when is apn_ambr->apnambrfordownlink set as 0xfe? Can you add a comment before this line?

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c (hunk 4 above)
Comment (ulaskozat) [reply lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c]: looks like we are not using them anywhere.

File: lte/gateway/c/core/oai/test/mme_app_task/test_EPSQualityOfService.cpp (hunk 9 above)
Comment (ulaskozat) [reply lte/gateway/c/core/oai/test/mme_app_task/test_EPSQualityOfService.cpp]: I do not see it updated.

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_emm_decode.cpp [hunk 14]
```diff
@@ -1,74 +0,0 @@
-/**
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_emm_decode.cpp]: why is this file removed?

Comment (ulaskozat) [reply lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_emm_decode.cpp]: okay ignore my comment.

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp [hunk 15]
```diff
@@ -0,0 +1,1355 @@
...
+  ASSERT_EQ((eps_qos.bitRatesExt2.maxBitRateForUL), 0);
+  ASSERT_EQ((eps_qos.bitRatesExt2.maxBitRateForDL), 0);
+  uint8_t eps_qos_buffersize_encoded1[10] = {0};
+  ASSERT_EQ(
+      encode_eps_quality_of_service(
+          &eps_qos, 0, eps_qos_buffersize_encoded1,
+          sizeof(eps_qos_buffersize_encoded1)),
+      sizeof(eps_qos_buffersize_encoded1));
+
+  ASSERT_EQ(sizeof(eps_qos_buffersize1), sizeof(eps_qos_buffersize_encoded1));
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp]: this is not needed as you are defining both arrays with the same size.

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp [hunk 16]
```diff
@@ -0,0 +1,1355 @@
...
+  ASSERT_EQ((eps_qos.bitRatesExt.maxBitRateForDL), 0);
+  ASSERT_EQ((eps_qos.bitRatesExt2.maxBitRateForUL), 0);
+  ASSERT_EQ((eps_qos.bitRatesExt2.maxBitRateForDL), 0);
+  uint8_t eps_qos_buffersize_encoded2[10] = {0};
+  ASSERT_EQ(
+      encode_eps_quality_of_service(
+          &eps_qos, 0, eps_qos_buffersize_encoded2,
+          sizeof(eps_qos_buffersize_encoded2)),
+      sizeof(eps_qos_buffersize_encoded2));
+  ASSERT_EQ(sizeof(eps_qos_buffersize2), sizeof(eps_qos_buffersize_encoded2));
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp]: please remove all similar lines, you do not need to compare sizes of arrays that you are defining.

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp [hunk 17]
```diff
@@ -0,0 +1,1355 @@
...
+TEST_F(NASEncodeDecodeTest, TestDecodeEncodeEPSQoS) {
+  EpsQualityOfService eps_qos = {0};
+
+  uint8_t eps_qos_buffersize1[] = {0x09, 0x09, 0x1c, 0x1c, 0x1c,
+                                   0x1c, 0x00, 0x00, 0x00, 0x00};
+  ASSERT_EQ(
+      decode_eps_quality_of_service(
+          &eps_qos, 0, eps_qos_buffersize1, sizeof(eps_qos_buffersize1)),
+      sizeof(eps_qos_buffersize1));
+  ASSERT_EQ((eps_qos.bitRates.maxBitRateForUL), 28);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp:39]: what about the other fields of eps_qos?

**Review by ulaskozat — CHANGES_REQUESTED** [review]
Please see additional comments.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c [hunk 18]
```diff
@@ -113,52 +118,120 @@ void bit_rate_value_to_eps_qos(
     ApnAggregateMaximumBitRate* apn_ambr, uint64_t ambr_dl, uint64_t ambr_ul) {
   uint64_t ambr_dl_kbps = ambr_dl / 1000;  // ambr_dl is expected in bps
   uint64_t ambr_ul_kbps = ambr_ul / 1000;  // ambr_ul is expected in bps
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: should not we now also using units explicitly in the function call?

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c [hunk 19]
```diff
@@ -228,8 +278,23 @@ status_code_e qos_params_to_eps_qos(
...
+          } else if ((mbr_dl_kbps > 500000) && (mbr_dl_kbps <= 1500000)) {
+            eps_qos->bitRatesExt2.maxBitRateForDL =
+                ((mbr_dl_kbps - 500000) / 10000) + 61;
+          } else if ((mbr_dl_kbps > 1500000) && (mbr_dl_kbps <= 10000000)) {
+            eps_qos->bitRatesExt2.maxBitRateForDL =
+                ((mbr_dl_kbps - 1500000) / 100000) + 161;
+          }
         }
       }
+
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c]: what about gbr values?

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 12 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c:168]: Comment updated.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 13 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c:181]: Ran make format_core to format the code.

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp (hunk 15 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp]: Updated

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp (hunk 16 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp]: Updated

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp (hunk 17 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/test/mme_app_task/test_mme_app_nas_encode_decode.cpp:39]: Updated

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 18 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: Updated

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c (hunk 19 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c]: Updated

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 11 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: Ran "make format_core" to format the code.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 10 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: Updated

File: lte/gateway/c/core/oai/test/mme_app_task/test_EPSQualityOfService.cpp (hunk 9 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/test/mme_app_task/test_EPSQualityOfService.cpp]: Code removed

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 8 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c:232]: Ran "make format_core" to format the code.

File: lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c (hunk 7 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c]: Ran "make format_core" to format the code.

File: lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c (hunk 4 above)
Comment (Vyas20) [reply lte/gateway/c/core/oai/tasks/nas/ies/EpsQualityOfService.c]: Removed  before-mentioned functions.

**Comment by ulaskozat** [comment]
merge conflicts

**Comment by ardzoht** [comment]
@Vyas20 Can you please pull latest master and rebase your changes on top of it? Let's unblock this review

**Comment by Vyas20** [comment]
> @Vyas20 Can you please pull latest master and rebase your changes on top of it? Let's unblock this review

Merge conflicts resolved.

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by ulaskozat** [comment]
Please mark as ready to merge. I see one of the linters failing about docs. Probably you need to rebase and push again.
